algorithms/ar-expsq/csharp/ExponentiationBySquaring.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+public class ExponentiationBySquaring
+{
+	public static double exponentiationBySquaring(double base_num, double exponent){
+		if(exponent == 0)
```

Comment:
Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6352723)